### PR TITLE
Add copy button to Recent Upload Item

### LINKED ISSUE #349: Enhancement: Ability to copy link from recent uploads
Add behavior where link will be copied on long press on it.

Or add a separate button to copy the link (having a separate button is a much better UX).

### PR DESCRIPTION
**Pull request by rod41732**
Fix #349 
- add button to allow user to copy image URL to clipboard in recent uploads item.

### CODE REVIEW

**Comment by rod41732** [comment]
Screenshots:
- [Situation: item with deletion links](https://user-images.githubusercontent.com/26536811/196932405-692fa45d-789e-41db-99b3-e7aa26762554.png)
- [Situation: item without deletion links](https://user-images.githubusercontent.com/26536811/196932470-057231aa-2c25-4e95-9327-272d6940de97.png)

File: app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt [hunk 1]
```diff
@@ -24,6 +30,16 @@ class RecentUploadsAdapter : ListAdapter<RecentUpload, RecentUploadsAdapter.Uplo
...
+            with(copyButton) {
+                setOnClickListener {
+                    val clipData = ClipData.newPlainText("Image URL", item.imageUrl)
+                    context.getSystemService<ClipboardManager>()?.setPrimaryClip(clipData)
+                    // show copied toast only on Android < 13
+                    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S_V2) {
+                        Toast.makeText(context, context.getString(R.string.copied_image_url), Toast.LENGTH_SHORT).show()
+                    }
+                }
+            }
```
Comment (flex3r) [inline app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt:42]: I don't like the additional nesting just for the button, I'd prefer something like:
```suggestion
            copyButton.setOnClickListener {
                val clipData = ClipData.newPlainText("Image URL", item.imageUrl)
                val context = it.context
                context.getSystemService<ClipboardManager>()?.setPrimaryClip(clipData)
                // show copied toast only on Android < 13
                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S_V2) {
                    Toast.makeText(context, context.getString(R.string.copied_image_url), Toast.LENGTH_SHORT).show()
                }
            }
```

File: app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt [hunk 2]
```diff
@@ -24,6 +30,16 @@ class RecentUploadsAdapter : ListAdapter<RecentUpload, RecentUploadsAdapter.Uplo
             image.load(item.imageUrl) {
                 scale(Scale.FILL)
             }
+            with(copyButton) {
+                setOnClickListener {
+                    val clipData = ClipData.newPlainText("Image URL", item.imageUrl)
+                    context.getSystemService<ClipboardManager>()?.setPrimaryClip(clipData)
+                    // show copied toast only on Android < 13
+                    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S_V2) {
```
Comment (flex3r) [inline app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt:38]: Android 13 is `Build.VERSION_CODES.TIRAMISU`, not `S_V2`

File: app/src/main/res/drawable/ic_copy.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="?attr/colorControlNormal"
```
Comment (flex3r) [inline app/src/main/res/drawable/ic_copy.xml:1]: I like to also format these kind of files for consistency and readability.

**Review by flex3r — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt (hunk 2 above)
Comment (rod41732) [reply app/src/main/kotlin/com/flxrs/dankchat/preferences/upload/RecentUploadsAdapter.kt:38]: I went pepega and forget the `=`, it's supposed to be `<=` FeelsDankMan

File: app/src/main/res/drawable/ic_copy.xml (hunk 3 above)
Comment (rod41732) [reply app/src/main/res/drawable/ic_copy.xml:1]: no idea why it didn't format the file FeelsDankMan

**Review by flex3r — APPROVED** [review]
LGTM image

**Pull request merged**